lib/sorcery/controller/submodules/jwt_auth.rb

```diff
@@ -0,0 +1,85 @@
+module Sorcery
+  module Controller
+    module Submodules
+      module JwtAuth
+        def self.included(base)
+          base.send(:include, InstanceMethods)
+        end
+
+        module InstanceMethods
+          # This method return generated token if user can be authenticated
+          def jwt_auth(*credentials)
+            user = user_class.authenticate(*credentials)
+            if user
+              now = Time.current
+              default_payload = {
+                  sub: user.id,
+                  exp: (now + 3.days).to_i,
+                  iat: now.to_i
+              }
+
+              payload = default_payload.merge Config.jwt_payload
+              
+              { Config.jwt_user_data_key => default_payload,
+                Config.jwt_auth_token_key => jwt_encode(payload) }
+            end
+          end
+
+          # To be used as a before_action.
+          def jwt_require_auth
+            binding.pry
+            @current_user = Config.jwt_set_user ? User.find(jwt_user_id) : jwt_user_data
+          rescue JWT::DecodeError => e
+            jwt_not_authenticated(message: e.message) && return
```

Comment:
Is trailing `return` call needed here?

Reply:
`jwt_not_authenticated` is currently only called when JWT is not decoded properly. We should also handle cases when JWT is properly decoded, but the user record is missing.